modules/api/src/main/java/org/apache/ignite/configuration/CoreLocalConfigurationModule.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.configuration;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
+
+import java.lang.annotation.Annotation;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.ignite.configuration.annotation.ConfigurationType;
+import org.apache.ignite.configuration.schemas.clientconnector.ClientConnectorConfiguration;
+import org.apache.ignite.configuration.schemas.network.NetworkConfiguration;
+import org.apache.ignite.configuration.schemas.rest.RestConfiguration;
+import org.apache.ignite.configuration.schemas.runner.NodeConfiguration;
+import org.apache.ignite.configuration.validation.ConfigurationModule;
```

Comment:
This package doesn't seem right to me

Reply:
Moved the class to `org.apache.ignite.configuration`. Do you think it's appropriate?

Reply:
I think that we have a problem here actually. ServiceLoader interfaces were supposed to be a part of internal modules "communication", but right now it is a "public" interface, which is not very convenient. I would argue that this interface should be moved into "configuration" module, the one that is "internal". This would complicate things a little bit.

Reply:
Moved it to package `org.apache.ignite.internal.configuration` of module `ignite-configuration`. Does this seem right?

Reply:
Yes, it does seem right now